### DISCUSSION TITLE
Relation to MLIR

### DISCUSSION THREAD

**Discussion by admercs** [openxla/xla#13] [Ideas]
Hi Everyone,

I was wondering how this project sees itself in regard to the MLIR project and its frontends such as [Polygeist](https://github.com/llvm/Polygeist). I see MLIR and XLA as intending to solve the same problems at different levels of abstraction, so I think this is important to consider. Are there plans for XLA to support MLIR?

Warm and Fuzzy Regards,

Adam

**Comment by joker-eph** [comment]
Hi,

MLIR solves quite a different problem than XLA: it attempts to makes rapid prototype of compilers straightforward, while allowing to bring them to production system as well. It also intends to create an ecosystem of compiler components  that can be shared and combined.

XLA on the other hand is an end-to-end "ready-to-use" ML compiler: we're working to define precisely the input graphs (with StableHLO) as well as the runtime API (likely PJRT). Behind these APIs, we're using more and more of MLIR. You may consider that the CPU/GPU flows inside XLA will be reusing all the possible components of MLIR to deliver the promise of the XLA API.
